cg/test/management/subjects.api.spec.js

```diff
@@ -194,4 +198,168 @@ describe('List subjects that have given consent', () => {
       ])
     );
   });
+
+  it('should not allow a page query with a negative page number', async () => {
+    //WHEN
+    const managementToken = await managementJWT.sign({ id: 1 });
+
+    const res = await fetch('/api/management/subjects/list?page=-1', {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${managementToken}`
+      }
+    });
+
+    //THEN
+    expect(res.ok).toBeFalsy();
+    expect(res.status).toBe(BadRequest.StatusCode);
+    expect(await res.json()).toEqual(
+      expect.objectContaining({
+        message: 'child "page" fails because ["page" must be a positive number]',
+        validation: {
+          source: 'query',
+          keys: ['page']
+        }
+      })
+    );
+  });
+
+  it('should not allow a page query with the zero page number', async () => {
+    //WHEN
+    const managementToken = await managementJWT.sign({ id: 1 });
+
+    const res = await fetch('/api/management/subjects/list?page=0', {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${managementToken}`
+      }
+    });
+
+    //THEN
+    expect(res.ok).toBeFalsy();
+    expect(res.status).toBe(BadRequest.StatusCode);
+    expect(await res.json()).toEqual(
+      expect.objectContaining({
+        message: 'child "page" fails because ["page" must be a positive number]',
+        validation: {
+          source: 'query',
+          keys: ['page']
+        }
+      })
+    );
+  });
+
+  it('should not allow a page query with a page number too big', async () => {
+    //WHEN
+    const managementToken = await managementJWT.sign({ id: 1 });
+
+    const res = await fetch('/api/management/subjects/list?page=99999999999999', {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${managementToken}`
+      }
+    });
+
+    //THEN
+    expect(res.ok).toBeTruthy();
+    expect(res.status).toBe(200);
+    expect(await res.json()).toEqual(
+      expect.objectContaining({
+        error: 'page number too big, maximum page number is 1'
+      })
+    );
+  });
+
+  it('should not allow a page query without an integer page number', async () => {
+    //WHEN
+    const managementToken = await managementJWT.sign({ id: 1 });
+
+    const res = await fetch('/api/management/subjects/list?page=string', {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${managementToken}`
+      }
+    });
+
+    //THEN
+    expect(res.ok).toBeFalsy();
+    expect(res.status).toBe(BadRequest.StatusCode);
+    expect(await res.json()).toEqual(
+      expect.objectContaining({
+        message: 'child "page" fails because ["page" must be a number]',
+        validation: {
+          source: 'query',
+          keys: ['page']
+        }
+      })
+    );
+  });
+
+  it('should allow a page query with a valid page number', async () => {
```

Comment:
We should also be validating that multi-page queries work. Check that you get served a page 2 that contains the expected items too

Reply:
Alright